From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: deepak1556 <hop2deep@gmail.com>
Date: Fri, 3 Sep 2021 18:28:51 -0700
Subject: fix: expose DecrementCapturerCount in web_contents_impl

This was made private in https://chromium-review.googlesource.com/c/chromium/src/+/2807829 but
we invoke it in order to expose contents.decrementCapturerCount([stayHidden, stayAwake])
to users. We should try to upstream this.

diff --git a/content/browser/web_contents/web_contents_impl.h b/content/browser/web_contents/web_contents_impl.h
index 78cdf5002d7b9b90fd152faa2b4fcd6f0159643e..f4f30aa6ad9b2196604624131e7de44d1d4f769b 100644
--- a/content/browser/web_contents/web_contents_impl.h
+++ b/content/browser/web_contents/web_contents_impl.h
@@ -1828,7 +1828,7 @@ class CONTENT_EXPORT WebContentsImpl : public WebContents,
   // IncrementCapturerCount() is destructed.
   void DecrementCapturerCount(bool stay_hidden,
                               bool stay_awake,
-                              bool is_activity = true);
+                              bool is_activity = true) override;
 
   // Calculates the PageVisibilityState for |visibility|, taking the capturing
   // state into account.
diff --git a/content/public/browser/web_contents.h b/content/public/browser/web_contents.h
index cd1b44d4ff5ce8924749ba9e41b3f599108bb8fd..72bc509b0506fafd026aa858864623886142b5f0 100644
--- a/content/public/browser/web_contents.h
+++ b/content/public/browser/web_contents.h
@@ -673,6 +673,10 @@ class WebContents : public PageNavigator,
       bool stay_awake,
       bool is_activity = true) = 0;
 
+  virtual void DecrementCapturerCount(bool stay_hidden,
+                                      bool stay_awake,
+                                      bool is_activity = true) = 0;
+
   // Getter for the capture handle, which allows a captured application to
   // opt-in to exposing information to its capturer(s).
   virtual const blink::mojom::CaptureHandleConfig& GetCaptureHandleConfig() = 0;
